### PR TITLE
Table widget modification

### PR DESCRIPTION
**Pull request by SebastienPeillet**

### CODE REVIEW

File: sammo.py [hunk 1]
```diff
@@ -6,12 +6,14 @@
 import os.path
 
 from qgis.PyQt.QtGui import QKeySequence
-from qgis.PyQt.QtWidgets import QToolBar, QShortcut
+from qgis.PyQt.QtCore import QObject
```
Comment (pblottiere) [inline sammo.py]: import order

**Comment by SebastienPeillet** [comment]
@pblottiere I pushed new commits because I was testing undo/redo mechanism with a uncustomized QGIS version, but `Ctrl+Z` shortcut is set on the undo action of the numerisation toolbar, and it doesn't exists in SAMMO light mode...

So I remove the existing shortcuts to avoid shorcut overload and I re-add new shortcuts.

I also try to add the new shortcuts only if the numerisation toolbar does not exist, but we only know it after initialisation completion, because the plugin is loadded before the customization file. We could create the shorcuts after iface.initializationCompleted signal emission but then if the plugin is reloaded, the shorcuts will not be created, so let's keep it simple :)

File: sammo.py [hunk 2]
```diff
@@ -202,6 +219,27 @@ def onCreateSession(self, sessionDirectory: str) -> None:
...
+                QTableView, "mTableView"
+            ):
+                self.iface.setActiveLayer(self.session.environmentLayer)
+            elif new == self.tableDock.widget().tables["Sightings"].findChild(
+                QTableView, "mTableView"
+            ):
+                self.iface.setActiveLayer(self.session.sightingsLayer)
+
+    def undo(self):
+        print(self.iface.activeLayer(), "test")
```
Comment (pblottiere) [inline sammo.py:237]:
```suggestion
```

**Comment by SebastienPeillet** [comment]
@pblottiere I fixed the crashes we met (with the _toDoInsideLoop message) in the 60b075a commit.

The table are guilty, in some way, the table dialogs are not destroy when the layer is removed from the project. So I clean them when these layers are removed. It prevents crash when closing the project or closing QGIS, but there still some side effect from the tableDock.

For example, it's not possible to create a join with the project layers, QGIS blocks on a QMutexLocker when updating fields (i'm not very aware of lock uses...). So to do the join in #132, I have to clone the project layers and use the clones to do the join.

**Comment by pblottiere** [comment]
> I  fixed the crashes we met (with the _toDoInsideLoop message) in the 60b075a commit.

Nice :tada:!

**Comment by pblottiere** [comment]
Good thanks :+1:! And ctrl-z is working too :).

**Pull request merged**